lib/v2/screens/sign_up/create_username/usecases/create_username_usecase.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:async/async.dart';
+import 'package:seeds/utils/string_extension.dart';
+import 'package:seeds/v2/datasource/remote/api/signup_repository.dart';
+import 'package:seeds/v2/screens/sign_up/create_username/mappers/create_username_mapper.dart';
+import 'package:seeds/v2/screens/sign_up/viewmodels/bloc.dart';
+import 'package:seeds/v2/screens/sign_up/viewmodels/states/create_username_state.dart';
+
+class CreateUsernameUseCase {
+  CreateUsernameUseCase({required SignupRepository signupRepository}) : _signupRepository = signupRepository;
+
+  final SignupRepository _signupRepository;
+
+  Stream<SignupState> validateUsername(SignupState currentState, String username) async* {
```

Comment:
Use cases are here to interact with the repositories. 

Use cases normally return `Result` values. 

The Bloc is in charge of `yield currentState.copyWith(createUsernameState: CreateUsernameState.loading(currentState.createUsernameState));`

`The state should never be passed into the use cases.`
Some use cases can be reused, if we need states inside the use cases then they cant be reused. 
It makes it more difficult to test uses cases because now you need to mock the state thats being passed in. 

The bloc is the one in charge of orchestration.

Reply:
They can be functions in the bloc too but I have been doing this for all the screens I have worked on. Let's address this in a separate PR.